addons/mail/static/src/core/public_web/discuss.js

```diff
@@ -67,6 +68,22 @@ export class Discuss extends Component {
             },
             { capture: true }
         );
+        if (this.store.inPublicPage) {
+            threadActionsRegistry.remove("close");
```

Comment:
Recently guests can have discuss sidebar:
https://github.com/odoo/odoo/pull/171016

So the close button should be disabled only if there's "no sidebar". The close button is important to go back to conversation selection in mobile!